kratos/sources/gid_io.cpp

```diff
@@ -0,0 +1,26 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Angel Celigueta
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/gid_io.h"
+
+namespace Kratos
+{
+    int GidIOBase::msLiveInstances = 0;
```

Comment:
how does this work if you have multiple instances or multiple classes deriving from the `GidIOBase`?

Did you try with the one where you have problems?

Reply:
Hi @philbucher . The plan is that there's only one copy of `msLiveInstances` , for all instances of GidIO, even with different template arguments (different classes deriving from GidIOBase). The fact of `msLiveInstances` being `static` and belonging to GidIOBase makes it unique. 
The problem might come when GidIO is derived, if constructors of GidIODerived do not call the constructor of GidIO...
Other problems can appear it two instances modify 'msLiveInstances' simultaneously from different threads.

Reply:
I tried the implementation last week (linux), and the problem had gone. I will run it now again.

Reply:
It runs fine.